src/hooks/useTokenAmountToUsd.js

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useState } from 'react'
+import { captureException } from '@sentry/browser'
+
+import { useCourtConfig } from '../providers/CourtConfig'
+import { useUniswapAnjPrice } from './useUniswapAnjPrice'
+import { bigNum, formatUnits } from '../lib/math-utils'
+import { getNetworkType } from '../lib/web3-utils'
+
+const API_BASE = 'https://api.0x.org'
+const PRECISION = bigNum(10).pow(18)
+const SELL_TOKEN = 'USDC'
+
+/**
+ * Convert ANJ amount into USD price
+ * @param {BigNum} amount The amount to convert to USD
+ * @returns {String} The amount value in USD
+ */
+export function useANJAmountToUsd(amount) {
+  const { anjToken } = useCourtConfig()
+  const anjPrice = useUniswapAnjPrice()
+
+  if (anjPrice === 0) {
+    return '-'
+  }
+
+  return convertAmount(amount, anjPrice, anjToken.decimals)
+}
+
+/**
+ * Convert a token into a USD price
+ *
+ * @param {String} symbol The symbol of the token to convert from.
+ * @param {Number} decimals The amount of decimals for the token.
+ * @param {BigNumber} amount The amount to convert into USD.
+ * @returns { Number } The balance value in USD
+ */
+export function useTokenAmountToUsd(symbol, decimals, amount) {
+  const [amountInUsd, setAmountInUsd] = useState('-')
+  useEffect(() => {
+    let cancelled = false
+
+    if (getNetworkType() !== 'main') {
+      return
+    }
+
+    const fetchPrice = async () => {
+      try {
+        const res = await fetch(
+          `${API_BASE}/swap/v0/prices?sellToken=${SELL_TOKEN}`
+        )
+        const prices = await res.json()
+
+        if (cancelled || !amount || !prices?.records?.length) {
+          return
+        }
+
+        const priceRecord = prices.records.find(
+          price => price.symbol === symbol
+        )
+        if (!priceRecord) {
+          return
+        }
+
+        const convertedAmount = convertAmount(
+          amount,
+          priceRecord.price,
+          decimals
+        )
+        setAmountInUsd(convertedAmount)
+      } catch (err) {
+        console.error(`Could not fetch ${symbol} USD price`, err)
+        captureException(err)
+      }
+    }
+
+    fetchPrice()
+
+    return () => {
+      cancelled = true
+    }
+  }, [amount, decimals, symbol])
+
+  return amountInUsd
+}
+
+function convertAmount(amount, price, decimals) {
+  const rate = price * PRECISION
```

Comment:
Ah interesting; if `PRECISION` is a `BigNumber`, how are we using normal operators? AFAIK we're not using native `BigInts` right?

Reply:
Ah when using a BigNumber on a native operation it's converted to its primitive value via `valueOf`.